### ISSUE TITLE
undefined method `id' for nil:NilClass

### ISSUE THREAD

**Issue by JoseTomasSoto** [IIC2143-2021-2/Syllabus#127]
Hola! 
En rails cree mi modelo como se indica en la ayudantía, pero me aparece que no hay id.
Cómo lo añado? Tenía entendido que el sistema se encargaba de eso.
Gracias!

**Comment by meretamal** [comment]
Hola 😁, el error que mencionas no se debe que no se está agregando la columna `id` a tu tabla, sino que el objeto que estás utilizando para acceder a su `id` está retornando `nil` (similar al `None` de python o `null` de JavaScript), entonces te sale que `nil` no tiene la propiedad `id`.

Puedes mandar un _snippet_ de tu código para ver qué puede estar pasando?

**Comment by JoseTomasSoto** [comment]
![Imagen1](https://user-images.githubusercontent.com/69685161/135955521-f545f4f3-ee7d-477f-a6eb-6bd424fbb92c.png)

Holaa! Supongo que te refieres a un fragmento del código.
La foto es lo que esta en migrate.

El schema es así 
ActiveRecord::Schema.define(version: 2021_10_04_175935) do

  # These are extensions that must be enabled in order to support this database
  enable_extension "plpgsql"

  create_table "users", force: :cascade do |t|
    t.string "name"
    t.string "username"
    t.string "email"
    t.string "password"
    t.string "photo"
    t.string "admin"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end

end

**Comment by JoseTomasSoto** [comment]
Efectivamente no tiene la propiedad id, pero en la ayudantía tampoco la mencionan explícitamente, debo hacerlo?

**Comment by JoseTomasSoto** [comment]
![image](https://user-images.githubusercontent.com/69685161/135955888-d10489ba-0558-43f1-91ee-9600918277dc.png)

Este es el error en rails

**Comment by OrregoSilva** [comment]
Probablemente la variable `@user` es la que vale `nil`, esa variable la tienes definida en el controlador?

**Comment by JoseTomasSoto** [comment]
Sí, debería, seguí al pie de la letra la ayudantía, gracias de antemano por su disposición!

class UsersController < ApplicationController
    def new
        @user = User.new
    end 

    def create
        @users_params = params.require(:user).permit(:name, :admin, :email, :photo, :password, :username)
        @user = User.create(@users_params)

        if @user.save
            redirect_to users_index_path, notice: 'User successfully created'
        else
            redirect_to users_index_path, notice: 'Error while creating the game'
        end
    end


    def index
        @users = User.all
    end

    def show
        @user = User.find(params[:id])
    end

    def edit
        @user = User.find(params[:id])
    end

    def update
        @user = User.find(params[:id])
        @users_new_params = params.require(:user).permit(:name, :admin, :email, :photo, :password, :username)
        if @user.update(@users_new_params)
            redirect_to users_index_path, notice: 'User edites'
        else
            redirect_to users_index_path, notice: 'User edit failed'
        end
    end
end

**Comment by meretamal** [comment]
Yap, viendo tu código me doy cuenta que estás intentando usar en la vista `index.html`la variable `@user`, que no está definida en el método `index` controlador. Por otro lado, sí está definida la variable `@users` y según lo veo de la foto que mandaste de tu error, en esa vista se está iterando sobre `@users` (con un bloque `each`) y se define la variable `user`, por lo que en verdad deberías usar `user` en esa línea, no `@user`.

Ahora, ojo que para manejar el crud de usuarios se recomienda user la gema `devise`, ya que también implementa muchas otras cosas, como manejo de sesión, _hashing_ de contraseñas, etc (se hizo una ayudantía de esto mismo)

**Comment by JoseTomasSoto** [comment]
Perfecto! Ahí se arregló, voy a implementar device igual. Gracias!
